grc/gui/ActionHandler.py

```diff
@@ -561,6 +561,25 @@ def flow_graph_update(fg=flow_graph):
                 Preferences.add_recent_file(file_path)
                 main.tool_bar.refresh_submenus()
                 main.menu_bar.refresh_submenus()
+        elif action == Actions.FLOW_GRAPH_SAVE_A_COPY:
+            try:
+                if not page.get_file_path():
+                    Actions.FLOW_GRAPH_SAVE_AS()
+                else:
+                    dup_file_path = page.get_file_path()
+                    dup_file_name = '.'.join(dup_file_path.split('.')[:-1]) + "_copy" # Assuming .grc extension at the end of file_path
+                    dup_file_path_temp = dup_file_name+'.grc'
+                    count = 1
+                    while (os.path.exists(dup_file_path_temp)):
+                        dup_file_path_temp = dup_file_name+'('+str(count)+').grc'
+                        count += 1
+                    ParseXML.to_file(flow_graph.export_data(), dup_file_path_temp)
+                    Messages.send("Successfully copied to " + dup_file_path_temp)
+            except IOError:
+                Messages.send_fail_save("Can not create a copy of the flowgraph")
+        elif action == Actions.FLOW_GRAPH_DUPLICATE:
+            curr_flow_graph = main.get_flow_graph()
+            main.new_page(flow_graph = curr_flow_graph)
```

Comment:
Here you need to save `flow_graph.export_data()`, then do the same as FLOW_GRAPH_NEW. that is `new_page()`  and update `flow_graph`. finally call `flow_graph.import_data(...)`